### DISCUSSION TITLE
Parent halo of satellite galaxies

### DISCUSSION THREAD

**Discussion by MarcoMarinucci** [galacticusorg/galacticus#414] [Q&A] [answered]
Hello,

I am working on a project where I am interested to know the total (central+satellite) number of galaxies that are within a halo of a given mass Mh. I'm running galacticus with a halo mass range [10^10-10^14] with 1000 trees per decade. I understand the in the output file this information is inside the 'parentIndex'. Now, my problem is that every halo of each tree has a 'nodeIndex'=1, so when I look at the 'parentIndex' of subhalos (selected using 'nodeIsIsolated'=0) I have a list of ones, and I don't have any other information about the specific halo that the sub halo belongs to. Is there a way I can obtain this information?
Just to repeat: I need the number of galaxies (central+satellite) that are inside a halo of mass Mh: N_gal (M_halo = M_h)

Thanks!

**Comment by abensonca** [comment, marked as answer]
You will need to also look at the index of the trees. As you've discovered, the `nodeIndex` values are unique only within each tree - so each tree will have a `nodeIndex=1` halo. If you also match nodes and their parents based on their tree index you'll be able to find just the satellites belonging to each parent halo.

There are a few ways to do this:

1. Add a `nodePropertyExtractor`:
```
<node property extractor value="indicesTree"/>
```
This will output a new property `mergerTreeIndex` which gives the index of each tree. This is the easiest approach.

2. Make use of the tree index information on each `Outputs/OutputN` group - see the top of page 15 of the [documentation](https://github.com/galacticusorg/galacticus/releases/download/bleeding-edge/Galacticus_Usage.pdf).

3. You can use the fact that each tree is contiguous in the output. So, all satellites associated with a given central appear immediately before that central in the output arrays.
